lib/at_cmd_parser/at_cmd_parser.c

```diff
@@ -52,6 +65,23 @@ static inline void skip_command_prefix(const char **cmd)
 	(*cmd)++;
 }
 
+static inline bool check_response_for_forced_string(const char *tmpstr)
+{
+	bool retval = false;
+
+	if (!strncmp(tmpstr, "+CGSN", AT_CMD_CGSN_LEN) ||
```

Comment:
This don't make sense for AT+CGSN.

AT+CGSN will only return the serial number without any notification identifier (+CGSN: ). For AT+CGSN=1, AT+CGSN=2, and AT+CGSN=3 the values are already returned with quotes. So they will be parsed as strings anyway.

Modem firmware version 1.2.1-57.rc